kolibri_dynamic_collections_plugin/assets/src/modules/collectionBase/index.js

```diff
@@ -139,24 +139,31 @@ export default {
       delete selectedNodeIdsByChannel[channelId];
       store.commit('SET_STATE', { ...store.state, selectedChannels, selectedNodeIdsByChannel });
     },
-    setNodeIncluded(store, { channelId, nodeId, included }) {
+    addSelectedNodes(store, { channelId, nodeIds }) {
       const selectedNodeIdsByChannel = { ...store.state.selectedNodeIdsByChannel };
-      const externalTagsByNode = { ...store.state.externalTagsByNode };
 
       const channelNodeIds = new Set(selectedNodeIdsByChannel[channelId]);
-      if (included) {
+      for (const nodeId of nodeIds) {
         channelNodeIds.add(nodeId);
-      } else {
-        channelNodeIds.delete(nodeId);
-        delete externalTagsByNode[nodeId];
       }
       selectedNodeIdsByChannel[channelId] = Array.from(channelNodeIds);
 
-      store.commit('SET_STATE', { ...store.state, externalTagsByNode, selectedNodeIdsByChannel });
+      store.commit('SET_STATE', { ...store.state, selectedNodeIdsByChannel });
     },
-    setExternalTagsForNode(store, { nodeId, tags }) {
+    removeSelectedNode(store, { channelId, nodeId }) {
+      const selectedNodeIdsByChannel = { ...store.state.selectedNodeIdsByChannel };
+
+      const channelNodeIds = new Set(selectedNodeIdsByChannel[channelId]);
+      channelNodeIds.delete(nodeId);
+      selectedNodeIdsByChannel[channelId] = Array.from(channelNodeIds);
```

Comment:
Maybe it's just me, but it seems like constructing a set just to remove a single value from an array is excessive. I'd think the following would be more efficient:

```
const channelNodeIds = selectedNodeIdsByChannel[channelId].filter(e => e !== nodeId)
selectedNodeIdsByChannel[channelId] = channelNodeIds
```

Alternatively, why not just store the nodes as a set?

Reply:
> Maybe it's just me, but it seems like constructing a set just to remove a single value from an array is excessive. I'd think the following would be more efficient:
> 
> ```
> const channelNodeIds = selectedNodeIdsByChannel[channelId].filter(e => e !== nodeId)
> selectedNodeIdsByChannel[channelId] = channelNodeIds
> ```

Yeah, that's a good point. Your one-liners there are simpler, even! I'll change that in a moment.

> Alternatively, why not just store the nodes as a set?

Alas, Vue's stores have this whole thing where data is mutable (distinct from Redux where everything is immutable) and the system magically handles mutations. But Map and Set aren't very well supported in this version. I don't really understand that in much detail, though. I already have three JavaScript state frameworks in my head and this one looks like all three of them and it freaks me out. So, uh, maybe I should just avoid Set altogether for consistency :)